src/substrait/textplan/RelationData.h

```diff
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include "substrait/proto/algebra.pb.h"
+
+struct RelationData {
+  RelationData(
+      const ::substrait::proto::Rel* relation,
+      const ::google::protobuf::Message* origin)
+      : protoAddr(relation), originProtoAddr(origin) {
+    continuingPipeline = nullptr;
+  };
+
+  const ::substrait::proto::Rel* protoAddr;
+  const ::google::protobuf::Message* originProtoAddr;
+
+  const ::substrait::proto::Rel* continuingPipeline;
+  std::vector<const ::substrait::proto::Rel*> newPipelines;
```

Comment:
Comments would be good here.  Just glancing at this I have no idea what any of these fields could mean.

Reply:
Updated.  This all will changes in the first parser PR where the underlying structure first has to handle both parsing and converting.